README.md

```diff
@@ -71,6 +71,11 @@ set -g @matryoshka_inactive_status_style 'fg=colour245,bg=colour238'
 # name of the option for the style of the status line
 # set if you rely on something other than the default 'status-style' option for it
 set -g @matryoshka_status_style_option 'my-status-style'
+
+# you can also set dynamic inactive status styles which
+# will be ran when the session becomes inactive
+set -g @matryoshka_dynamic_inactive_style 'on'
+set -g @matryoshka_inactive_status_style '$(tmux show-option -gqv window-status-format)'
```

Comment:
Would it still work for catppuccin users to instead define `inactive_status_style_option`, matching the existing `status_style_option`? It would simplify things quite a bit logic-wise. 👍 

If not, could you explain why and perhaps provide some basic requirements you have (agnostic from catppuccin) that matryoshka currently does not satisfy with regards to this problem you are having?

I am generally against passing arbitrary commands as configuration options since the possibility of shooting yourself in the foot is high!

Reply:
Apologies for the late response. The issue arises with how (at-least catppuccin in particular) handles the styling. It (and I assume a lot of other status-bar plugins) use very complex status-formats to handle their looks.

This, for instance, is catppuccin's window-status-format:
```
#[fg=#313244,bg=default,nobold,nounderscore,noitalics]#[fg=#cdd6f4,bg=#313244]#W#{?window_activity_flag, 󱅫,}#{?window_bell_flag, 󰂞,}#{?window_silence_flag, 󰂛,}#{?window_active,,}#{?window_last_flag,,}#{?window_marked_flag, 󰃀,}#{?window_zoomed_flag, 󰁌,}#[fg=#a9a9a9,bg=#313244,nobold,nounderscore,noitalics] █#[fg=#313244,bg=#a9a9a9]#I#[fg=#a9a9a9,bg=default] 
```

There's a lot going on here, from icons, to special nerdfont characters that allow smooth looking status bars (have a look at catppuccin's status bar to see what I mean).

While of course #1 can be used with the whole string, but it would look a little janky, and maybe a little too hacky. The status bar is not configured this way, but rather there are certain configuration values (for instance, @catppuccin_window_status_color) that are then used to generate the status-bar.

My solution offers a command-based approach for those wanting to keep their config a little cleaner. Another way I can see this being used is with a `echo <current window status> | sed <what needs to be changed`, to surgically modify one value that the users intends to change.

I will re-iterate that this is needed because complex status-bars are quite large "magic strings" that, while they can be copied, it would look a little odd.

Might I also recommend adding the ability for users to just run a custom command instead of the current schema (setting one option to one value/command), as that would allow complete customizability in cases of complex configurations.

Have a good day!

Reply:
I understand that catpuccin's window status format is highly complex and dynamic, but the option containing this dynamic status is always named `window-status-format`, correct? That's what the PR change also seems to suggest (please tell me if I'm wrong).

If that's the case, then a simple override of the option name for the inactive status style (much like we already have one for the status style) should work. Keep in mind that #1's goal was to "pick a particular style or config option in another Tmux style plugin to change when nesting", which sounds very similar to what we are trying to accomplish here.

If it's not, can you help me figure out what I'm missing? Is the example added in the README illustrative of what you need to configure to have catpuccin work?

Reply:
Not quite. The issue is about being able to run a command to change the window-status-format, rather than a hard-coded value. I could do `set -g @matryoshka_inactive_status_style '<long catppuccin string>'`, which would work fine, but is prone to two issues in my opinion.

one, it spams my config with magic strings.
two, if at any moment I change one of the settings in catppuccin, it could completely change my magic string, thereby needing to fetch it again and update the config.

instead, what I'm suggesting is the ability to run a command (or, "eval" a value), get its output, and set the inactive value accordingly. In the README, I have provided an example which I believe should be adequate to understand this problem (`set -g @matryoshka_inactive_status_style '$(tmux show-option -gqv window-status-format)'`).

If you still have any questions I'd love to hear them :)

Reply:
> p.s another example is, say, you have an external variable, let's call it $MY_VAR.
we can then do set -g @matryoshka_inactive_status_style ${MY_VAR}. and then we can dynamically change $MY_VAR with external scripts, and it would be dynamically updated without having to reload the config.

Thank you for this example, I understand a little better now.

I see that the problem is fundamentally different from #1, since in the latter the plugin seemed to move the tmux source-of-truth option away from the default `status-style`, requiring a tweak on what option would be updated. Instead, the catppuccin plugin features this `window-status-format` that _contains_ the status style value you would like to set, but is not the option that is used to actually render the status bar. Correct?

If I fully understand now, I think we can still do without the arbitrary command invocation. It looks like what you're essentially doing with the PR change is specifying a source of truth for the inactive status style that is bound to a dynamic tmux option. What I would do in this case is to allow the user to specify a tmux source-of-truth option for the inactive status style by specifying its name in the config.

I still didn't think of a way to properly implement this, but please let me know if it all makes sense to you. Moreover, do you think having the same option for the status style would be beneficial? Or is `set -u status-style` enough to restore the dynamic active status style for catppuccin users as well?

Reply:
I believe `window-status-format` is universal to tmux, but that's besides the point.

You are sort of correct, at least for my case. `window-status-format` is the inactive style of windows. In my case, it shows every window in the status bar as a gray circle, while `window-status-current-format` is a green circle.
![image](https://github.com/user-attachments/assets/53b6760b-655e-4e01-9745-87f2f6135099)

The style text itself it complex, and I just want matryoshka to use the exact same style for `window-status-current-format` as `window-status-format`. So yes, in this case, doing (pseudocode) `window-status-current-format = window-status-format` would indeed work, but if we only implement it this way, dynamically modifying the content would no longer be possible. For instance, using `sed` to modify one color value in `window-status-current-format`'s text (changing #222222 to #333333 for instance).

Reply:
Ok, I see, thanks for bearing with me. I am not sure having an arbitrary command that the user is supposed to use along with `sed` to obtain the content of the disabled status line is the best solution, though.

Isn't there a way for catppuccin users to set a tmux option, say, `@catppuccin_status_line_bg_color`, and obtain the recoloring of the status line? If there is, I would be happy to implement the code that allows you to specify a tmux option to set and reset to a specific value when switching in and out of disabled mode. Something like:

```
set -g @matryoshka_inactive_tmux_option 'catppuccin_status_line_bg_color,gray'
```

Which would entail a backup of the original value of `catppuccin_status_line_bg_color` (say, `yellow`) before overwriting it with the value `gray` whenever tmux is disabled. When enabled again, we would use the backed up value (stored somewhere) to restore the original value.

You can also give a shot at the implementation instead, if you'd like!

In any case, let me know if this catppuccin option actually exists. I think it does judging by their README, but I am not sure if the program also needs some re-sourcing to have the option reflect on the status line, could you test that?

Finally, in the unfortunate case that we're gonna need to run some command to have catppuccin know of this option update, we might consider having direct integration with catppuccin by running the command ourselves, so that we can still avoid having the user specify that command externally through configuration.

Reply:
Unfortunately, no. For the vast majority of plugins, it's a set it and forget it. That is, once you set an `@option` and load the plugin, there's no changing it beyond reloading the entire configuration (or reloading the plugin).

Setting, in my case, `window-status-format` is a direct and immediate change, unlike `@catppuccin_<option>`.

As for the other concern of backing up the data, luckily there's no need. `-u` works fine!

Reply:
Hey @ItzYeho, sorry for getting back to this after weeks. I gave it some thought and I think the use case of replacing a value in the `status_style` with another (more often than not, a color) is something that can be made a first class citizen of the configuration. What I would like to do is the following:

* Introduce a new configuration option `@matryoshka_inactive_status_style_update_mode`, with two possible values: `overwrite` (the default value) and `replace`
* If `overwrite` is specified, we have the same behavior as before - `@matryoshka_inactive_status_style` is copied as is in the status style option
* If `replace` is specified, then a new configuration option `@matryoshka_inactive_status_style_replacements` will hold a list of tuples of the form
  ```
  before1,after1;before2,after2;...;beforeN,afterN
  ```
  The plugin can then take care of converting this high-level list into a `sed` command (or something of the sort) that replaces what the user intended to replace.

This way, I believe we achieve the functionality we are looking for while avoiding arbitrary command execution and keeping the configuration friendly for the user. What do you think?

Reply:
Actually, I kind of like that. It achieves the `sed` idea without the risk of running arbitrary commands. Quite neat! Only gripe I have with it is that it does not support external variables. But I don't think that's too big of a deal.

Reply:
I have replaced the commit and added a solution. Note this solution **does not account for escape characters**. That is, if the user wants to replace a semicolon or a colon, which are currently used as separators in their own right, they cannot do such. I'll leave an option here that does enable escaped characters, but it's not the prettiest.

```bash
#!/bin/bash
function split_replacements() {
    local input="$1"
    local delimiter="$2"
    local result=()
    local buffer=""
    local escape=0

    for (( i=0; i<${#input}; i++ )); do
        char="${input:$i:1}"
        if [[ "$escape" -eq 1 ]]; then
            # If the previous character was an escape character, add this char to the buffer
            buffer+="$char"
            escape=0
        elif [[ "$char" == "\\" ]]; then
            # If the current character is a backslash, the next character is escaped
            escape=1
        elif [[ "$char" == "$delimiter" ]]; then
            # If the current character is the delimiter and it's not escaped, we split
            result+=("$buffer")
            buffer=""
        else
            # Otherwise, add the character to the buffer
            buffer+="$char"
        fi
    done
    # Add the final buffer
    result+=("$buffer")
    
    echo "${result[@]}"
}

# Step 3: Process the REPLACEMENTS string
final_string="$original_string"

# Split by unescaped semicolons
pairs=$(split_replacements "$inactive_status_style_replacements" ';')

# Loop through each pair
for pair in $pairs; do
    # Split each pair by unescaped commas
    parts=$(split_replacements "$pair" ',')
    before=$(echo "$parts" | awk '{print $1}')
    after=$(echo "$parts" | awk '{print $2}')
    
    # Apply the replacement
    final_string="${final_string//${before}/${after}}"
done
```